### ISSUE TITLE
Unblock platform so exit signals will stop runner [PAXRUNNER-43]

### ISSUE THREAD

**Issue by ops4j-issues** [ops4j/org.ops4j.pax.runner#64]
**[David Leangen](https://ops4j1.jira.com/secure/ViewProfile.jspa?accountId=557058:51aa7020-bc9c-4e10-bb9f-bc5a45225f5b)** created **[PAXRUNNER-43](https://ops4j1.jira.com/browse/PAXRUNNER-43)** 

Line 244 of ProcessImpl is a problem:

process.waitFor();

This causes the Pax Runner process to block until the OSGi platform has completed it's work. The problem is that even when Pax Runner receives a shutdown signal from my daemon, the signal does not get passed to the platform process, so the platform continues to run indefinitely.

Not sure how to fix this, since I don't have all that much experience with threads and such, but I'll look into this to provide some suggestions.

---

**Fixed in:** 0.5.0
Votes: 0, Watches: 0

**Comment by ops4j-issues** [comment]
**[David Leangen](https://ops4j1.jira.com/secure/ViewProfile.jspa?accountId=557058:51aa7020-bc9c-4e10-bb9f-bc5a45225f5b)** commented

Ok... It looks like destroyFrameworkOnExit is already set up to handle proper termination, so I don't understand why it's not working...



My pax runner process gets send a TERM signal from the daemon system, but the process is not terminated, so something is wrong somewhere...



Any ideas?

**Comment by ops4j-issues** [comment]
**[David Leangen](https://ops4j1.jira.com/secure/ViewProfile.jspa?accountId=557058:51aa7020-bc9c-4e10-bb9f-bc5a45225f5b)** commented

Some more info...



I inserted some debug logs to test out what's happening here.



When I send the SIGINT (CTRL-C) from the shell, destroyFrameworkOnExit, and therefore so does process.destroy();



When my daemon sends the SIGTERM signal, the signal is not being caught. Normally, the JVM is supposed to treat SIGINT, SIGTERM, and SIGHUP the same way, so I'm not sure what's happening here...

**Comment by ops4j-issues** [comment]
**[Alin Dreghiciu](https://ops4j1.jira.com/secure/ViewProfile.jspa?accountId=557058:a786db11-2013-41fb-af1d-ca5eb8f74aa0)** commented

Replay fro David Leangen:



"Well... I've finally managed to get around the SIGTERM problem.



Turns out that it's due to something happening in the shell, not in the<br /> Java, but I still have no idea why. Guess I'll have to get deep into the<br /> shell mechanics to understand, but, as always, I don't have time.



So, I am able to get up and running with my daemon, so I'm ok with a<br /> release."
